### PR TITLE
Use MaterialUI instead of Bootstrap for Lesson Plan

### PR DESCRIPTION
**Pull request by kxmbrian**
Basically overhauled the whole component

### CODE REVIEW

File: client/app/lib/helpers/viewport.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Returns true if screen width is smaller than 768px
+ */
+function isScreenSm() {
+  const width = window.innerWidth
+    || document.documentElement.clientWidth
+    || document.body.clientWidth;
+
+  return width < 992;
```
Comment (allenwq) [inline client/app/lib/helpers/viewport.js]: Not sure why the number is 992, documentation says 768

**Comment by allenwq** [comment]
@kxmbrian As discussed:

- I strongly feel there's a need to let user to easily access the sidebar after they visit the lesson plan page >< 
- Render new/edit pages in the frontend and make lesson plan a sample component( Standardise forms / routers ) for future frontend development. Think I can help with this.

File: client/app/lib/helpers/viewport.js (hunk 1 above)
Comment (kxmbrian) [inline client/app/lib/helpers/viewport.js]: Updated this to `isScreenXs` which checks that if width < 768.

File: app/views/course/assessment/assessments/_assessment_lesson_plan_item.json.jbuilder [hunk 2]
```diff
@@ -1,3 +1,13 @@
 json.partial! 'course/lesson_plan/items/item.json.jbuilder', item: item
 
 json.lesson_plan_item_type @assessment_tabs_titles_hash[item.tab_id]
+json.item_path course_assessment_path(current_course, item)
+folder = @folder_loader.folder_for_assessment(item.id)
+if can?(:attempt, @assessment) && !folder.materials.empty?
+  json.materials do
```
Comment (weiqingtoh) [inline app/views/course/assessment/assessments/_assessment_lesson_plan_item.json.jbuilder]: I will need to use the `materials` json data - possible to factor that out as a separate partial?

Comment (kxmbrian) [inline app/views/course/assessment/assessments/_assessment_lesson_plan_item.json.jbuilder]: Have factored it out

File: client/app/bundles/course/lesson_plan/components/LessonPlanNav.jsx [hunk 3]
```diff
@@ -1,29 +1,144 @@
...
+      open: false,
+    });
+  }
+
+  /**
+   * This uses react-scroll's scrollspy logic to update the text on the milestone navigation
+   * button. The easiest way to do this is to make use of the higher order component,
+   * Helpers.Scroll. However, since the Popover Menu is not mounted unless the menu is open,
+   * the scroll listeners will be not mounted when the menu is closed if we choose to wrap
+   * MenuItems with Helpers.Scroll. To workaround this, we wrap empty dummy spans instead.
```
Comment (allenwq) [inline client/app/bundles/course/lesson_plan/components/LessonPlanNav.jsx]: @kxmbrian Sorry, I don't really get the problem here.

**Comment by allenwq** [comment]
@kxmbrian  As discussed, think it would be better to factor out the time formats helper into a central place and use it as the default display helper of the time for most of the cases.

BTW, I think there will be some timezone issues in frontend also, will discuss with you offline.

Other than that, LGTM.

File: app/views/course/assessment/assessments/_assessment_lesson_plan_item.json.jbuilder (hunk 2 above)
Comment (weiqingtoh) [inline app/views/course/assessment/assessments/_assessment_lesson_plan_item.json.jbuilder]: 👍 thanks @kxmbrian

**Pull request merged**